android/src/com/mapswithme/maps/intent/Factory.java

```diff
@@ -250,203 +110,88 @@ public MapTask process(@NonNull final Intent intent)
     }
   }
 
-  private static class HttpMapsIntentProcessor extends BaseOpenUrlProcessor
+  public static class HttpMapsIntentProcessor implements IntentProcessor
   {
-    @Override
-    public boolean isSupported(@NonNull Intent intent)
-    {
-      final String scheme = intent.getScheme();
-      final Uri data = intent.getData();
-      if (data == null || (!"http".equalsIgnoreCase(scheme) && !"https".equalsIgnoreCase(scheme)))
-        return false;
-      final String host = data.getHost();
-      return host.contains("google") || host.contains("2gis") || host.contains("openstreetmap");
-    }
-
-    @NonNull
-    @Override
-    MapTask createMapTask(@NonNull String uri)
-    {
-      return new OpenHttpMapsUrlTask(uri);
-    }
-  }
-
-  private static class OldLeadUrlIntentProcessor extends BaseOpenUrlProcessor
-  {
-    @Override
-    public boolean isSupported(@NonNull Intent intent)
-    {
-      final Uri data = intent.getData();
-
-      if (data == null)
-        return false;
-
-      String scheme = intent.getScheme();
-      String host = data.getHost();
-      if (TextUtils.isEmpty(scheme) || TextUtils.isEmpty(host))
-        return false;
-
-      return (scheme.equals("mapsme") || scheme.equals("mapswithme")) && "lead".equals(host);
-    }
-
-    @NonNull
-    @Override
-    MapTask createMapTask(@NonNull String uri)
-    {
-      return new OpenUrlTask(uri);
-    }
-  }
-
-  private static class OldCoreLinkAdapterProcessor extends DlinkIntentProcessor
-  {
-    private static final String SCHEME_CORE = "mapsme";
-
-    @NonNull
-    @Override
-    protected MapTask createTargetTask(@NonNull String url)
-    {
-      // Transform deeplink to the core expected format,
-      // i.e https://host/path?query -> mapsme:///path?query.
-      Uri uri = Uri.parse(url);
-      Uri coreUri = uri.buildUpon()
-                       .scheme(SCHEME_CORE)
-                       .authority("").build();
-
-      String query = coreUri.getLastPathSegment();
-      return BackUrlMapTaskWrapper.wrap(new OpenUrlTask(coreUri.toString()), url);
-    }
-
-    @Override
-    boolean isLinkSupported(@NonNull Uri data)
-    {
-      return true;
-    }
-
     @Nullable
     @Override
-    MapTask createIntroductionTask(@NonNull String url)
+    public MapTask process(@NonNull Intent intent)
     {
-      return null;
+      final Uri uri = intent.getData();
+      if (uri == null)
+        return null;
+      final String scheme = intent.getScheme();
+      if (!"http".equalsIgnoreCase(scheme) && !"https".equalsIgnoreCase(scheme))
+        return null;
+      final String host = uri.getHost();
+      if (host == null)
+        return null;
+      if (!host.contains("google") && !host.contains("2gis") && !host.contains("openstreetmap"))
```

Comment:
nit: Should we also process _ignore case_ like for schemes, or it's too much?

Reply:
I changed to "host = uri.getHost().toLower()".